src/Microsoft.DotNet.Cli.Utils/FrameworkDependencyFile.cs

```diff
@@ -45,6 +46,36 @@ public string GetNetStandardLibraryVersion()
                 ?.Version;
         }
 
+        public bool TryGetMostFitRuntimeIdentifier(IEnumerable<string> runtimeIdentifiers, out string mostFitRuntimeIdentifier)
+        {
+            mostFitRuntimeIdentifier = null;
+
+            IEnumerable<string> supportedRuntimeIdentifiers = runtimeIdentifiers.Distinct().Where(r => IsRuntimeSupported(r));
+
+            if (supportedRuntimeIdentifiers.Any())
+            {
+                mostFitRuntimeIdentifier = supportedRuntimeIdentifiers.Where(r => IsAllOtherNodeOnPathToRoot(r, supportedRuntimeIdentifiers)).First();
```

Comment:
What if you have two different candidate RIDs that both match but where neither is an ancestor of the other?  Is this possible?  An example could be `win10` and `win-x64` if those are both valid.  It looks like the code here would crash in this case (`First()` wouldn't find any elements).

Reply:
good point, it is actually a graph or tree. i always remember it is a tree. let me check

Reply:
it is a graph, this won't work